android/app/src/main/java/com/reactnativenavigation/modal/RnnModal.java

```diff
@@ -87,4 +105,17 @@ public void onBackPressed() {
     public void onDismiss(DialogInterface dialog) {
         ModalController.getInstance().remove();
     }
+
+    @Override
+    public void dismiss(){
+        super.dismiss();
+        if (mContext != null && !mContext.isFinishing()) {
+            mContext.runOnUiThread(new Runnable() {
+                @Override
+                public void run() {
+                    updateToolbar(mOriginalScreen, mOriginalToolbar);
```

Comment:
Great. What do you think about moving `updateToolbar` to `RnnToolbar` class? I didn't have time to check this, but I think it will be used from other places as well. 

Maybe something like this:

```
@UiThread
    public void updateToolbar(Screen screen) {
        setTitle(screen.title);
        setSupportActionBar();
        setStyle(screen);
        setupToolbarButtonsAsync(screen);
    }

    public void setSupportActionBar() {
        ((AppCompatActivity) getContext()).setSupportActionBar(this);
    }
```

Then, in `RnnModal` we would call `mToolBar.updateToolbar(mScreen);`
This also eliminates the need to keep a hard ref to `mOriginalToolbar`, a View that belongs to the previous Activity/Modal, which is a bad practice.

If we expose `setSupportActionBar` we also don't need to keep a ref to `mOriginalScreen` - seems like it simplifies the class a bit.